File: Lab-1/Task-3/GraphAlgorithms/GraphAlgorithms/Node.cs
namespace GraphAlgorithms;

public class Node<T>
{
    private T _value;
    private readonly HashSet<Node<T>> _neighbors;
    
    public Node(T value) {
        _value = value;
        _neighbors = new HashSet<Node<T>>();
    }

    public void connect(Node<T> node)
    {
        _neighbors.Add(node);
        node._neighbors.Add(this);
    }

    public T getValue() {
        return _value;
    }

    public IReadOnlySet<Node<T>> getNeighbors() {
        return _neighbors;
    }
}

File: Lab-1/Task-3/GraphAlgorithmsPackage/src/main/java/com/bibletoon/graphAlgorithms/Node.java
package com.bibletoon.graphAlgorithms;

import java.util.Collection;
import java.util.HashSet;
import java.util.Set;

public class Node<T> {
    private T value;
    private Set<Node<T>> neighbors;

    public Node(T value) {
        this.value = value;
        this.neighbors = new HashSet<>();
    }

    public void connect(Node<T> node) {
        this.neighbors.add(node);
        node.neighbors.add(this);
    }

    public T getValue() {
        return value;
    }

    public Collection<Node<T>> getNeighbors() {
        return neighbors;
    }
}